src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/UrlRewriter.java

```diff
@@ -110,30 +118,45 @@ public static UrlRewriter getDownloaderUrlRewriter(String configPath, Reporter r
 
   /**
    * Updates {@code authHeaders} using the userInfo available in the provided {@code urls}.
+   * Note that if the same url is present in both {@code authHeaders} and <b>download config</b>
+   * then it will be overridden with the value from <b>download config</b>.
    *
    * @param urls The input list of {@link URL}s. May be empty.
    * @param authHeaders A map of the URLs and their corresponding auth tokens.
    * @return A map of the updated authentication headers.
    */
   public Map<URI, Map<String, String>> updateAuthHeaders(
       List<URL> urls, Map<URI, Map<String, String>> authHeaders) {
-    ImmutableMap.Builder<URI, Map<String, String>> authHeadersBuilder =
```

Comment:
@keith ImmutableMap.Builder doesn't like duplicate entries, see https://stackoverflow.com/questions/19773911/why-does-immutableset-allow-duplicates-but-immutablemap-does-not